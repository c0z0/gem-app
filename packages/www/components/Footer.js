import React from 'react'
import styled from 'styled-components'

import Container from './Container'
import { Diamond } from './Svg'

const FooterContainer = styled(Container)`
  display: flex;
  justify-content: space-between;
  align-items: center;

  @media (${({ theme }) => theme.b.phoneOnly}) {
    flex-direction: column;
  }
`

const Border = styled.footer`
  background: #fafafa;
  border-top: solid 1px #eaeaea;
  padding: 48px 0;
  color: #cecece;
  position: absolute;
  right: 0;
  bottom: 0;
  left: 0;
`

const Logo = styled.div`
  display: flex;
  align-items: center;
`

const Title = styled.span`
  margin-left: 8px;
  font-weight: 400;
  margin-right: 20px;
  color: #cecece;
  font-size: 20px;
`

const StyledDiamond = styled(Diamond).attrs({
  alt: 'gem logo'
})`
  width: 40px;
  filter: grayscale(100%);
  opacity: 0.5;
`

const Link = styled.a`
  color: #cecece;
  text-decoration: none;
  transition: all 0.2s;

  &:hover {
    color: #484848;
  }
`

export default function Footer() {
  return (
    <Border>
      <FooterContainer>
        <Logo>
          <StyledDiamond />
          <Title>Gem</Title>
        </Logo>
        <p>
          Made by <Link href="https://cserdean.com">cserdean.com.</Link>
        </p>
      </FooterContainer>
    </Border>
  )
}
