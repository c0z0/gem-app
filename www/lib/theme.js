export default {
  main: '#75489B',
  mainGradient: 'linear-gradient(315deg,#75489b 0%,#a8708d 100%)',
  b: {
    phoneOnly: `max-width: 991px`,
    tabletUp: `min-width: 992px`,
  },
}
