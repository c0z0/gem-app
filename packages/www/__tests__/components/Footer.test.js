import React from 'react'
import { render } from 'react-testing-library'
import { ThemeProvider } from 'styled-components'
import theme from '../../lib/theme'
import 'jest-styled-components'

import Footer from '../../components/Footer'

test('Footer matches snapshot', () => {
  const { container } = render(
    <ThemeProvider theme={theme}>
      <Footer />
    </ThemeProvider>
  )

  expect(container).toMatchSnapshot()
})
