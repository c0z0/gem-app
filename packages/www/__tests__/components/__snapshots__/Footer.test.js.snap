// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer matches snapshot 1`] = `
.c1 {
  max-width: 900px;
  margin: 0 auto;
  padding: 0 12px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  background: #fafafa;
  border-top: solid 1px #eaeaea;
  padding: 48px 0;
  color: #cecece;
  position: absolute;
  right: 0;
  bottom: 0;
  left: 0;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c4 {
  margin-left: 8px;
  font-weight: 400;
  margin-right: 20px;
  color: #cecece;
  font-size: 20px;
}

.c3 {
  width: 40px;
  -webkit-filter: grayscale(100%);
  filter: grayscale(100%);
  opacity: 0.5;
}

.c5 {
  color: #cecece;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 0.2s;
  transition: all 0.2s;
}

.c5:hover {
  color: #484848;
}

@media (max-width:991px) {
  .c1 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <img
          class="c3"
          src="/static/diamond.svg"
        />
        <span
          class="c4"
        >
          Gem
        </span>
      </div>
      <p>
        Made by 
        <a
          class="c5"
          href="https://cserdean.me"
        >
          cserdean.me.
        </a>
      </p>
    </div>
  </div>
</div>
`;
